src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp

```diff
@@ -132,18 +133,36 @@ CHIP_ERROR GenericPlatformManagerImpl_POSIX<ImplClass>::_StartChipTimer(int64_t
 template <class ImplClass>
 void GenericPlatformManagerImpl_POSIX<ImplClass>::_PostEvent(const ChipDeviceEvent * event)
 {
+    LockEventQueue();
     mChipEventQueue.push(*event); // Thread safe due to ChipStackLock taken by App thread
```

Comment:
comment needs to be updated.

I believe we should have lock/unlock as RAII rather than C-style lock/unlock